test/node/stage_2.ts

```diff
@@ -0,0 +1,77 @@
+import { promises as fs } from 'fs'
+import { join } from 'path'
+import process from 'process'
+import { pathToFileURL } from 'url'
+
+import test from 'ava'
+import del from 'del'
+import { execa } from 'execa'
+import semver from 'semver'
+import tmp from 'tmp-promise'
+
+import { getLocalEntryPoint } from '../../node/formats/javascript.js'
+
+test('`getLocalEntryPoint` returns a valid stage 2 file for local development', async (t) => {
+  const { path: tmpDir } = await tmp.dir()
+
+  // This is a fake bootstrap that we'll create just for the purpose of logging
+  // the functions and the metadata that are sent to the `boot` function.
+  const printer = `
+    export const boot = async (functions, metadata) => {
+      const responses = {}
+
+      for (const name in functions) {
+        responses[name] = await functions[name]()
+      }
+
+      console.log(JSON.stringify({ responses, metadata }))
+    }
+  `
+  const printerPath = join(tmpDir, 'printer.mjs')
+
+  await fs.writeFile(printerPath, printer)
+  process.env.NETLIFY_EDGE_BOOTSTRAP = pathToFileURL(printerPath).toString()
+
+  const functions = [
+    { name: 'func1', path: join(tmpDir, 'func1.mjs'), response: 'Hello from function 1' },
+    { name: 'func2', path: join(tmpDir, 'func2.mjs'), response: 'Hello from function 2' },
+  ]
+
+  for (const func of functions) {
+    const contents = `export default () => ${JSON.stringify(func.response)}`
+
+    await fs.writeFile(func.path, contents)
+  }
+
+  const stage2 = getLocalEntryPoint(
+    functions.map(({ name, path }) => ({ name, path })),
+    {},
+  )
+  const stage2Path = join(tmpDir, 'stage2.mjs')
+
+  await fs.writeFile(stage2Path, stage2)
+
+  // In Node <14, we're not able to actually load the stage 2 because ESM is
+  // not supported. The best we can do is to naively look for some assignments.
+  if (semver.lt(process.version.slice(1), '14.0.0')) {
+    for (const func of functions) {
+      t.true(stage2.includes(`metadata["${func.name}"] = {"url":"${pathToFileURL(func.path).toString()}"}`))
+    }
+
+    return
+  }
+
+  const { stdout, stderr } = await execa('node', [stage2Path])
```

Comment:
my understanding would be that the code you are generating runs in deno? why are you executing node here?

Reply:
This code runs in Node as part of local development, called by the CLI.

Reply:
Called here: https://github.com/netlify/cli/blob/main/src/lib/edge-functions/proxy.js#L166-L180

Reply:
Hang on, I'm holding this wrong! This is called from Node but then actually runs in Deno. 🤯 

Let me change this.

Reply:
Done in https://github.com/netlify/edge-bundler/pull/126/commits/4fcd2c190c0aee5272029128b3b5b7e7363548a2. Thanks!